test/runtests.jl

```diff
@@ -490,4 +490,178 @@ end
+        @test_throws ArgumentError parse_azure_url(container="192.168.5.4")
+
+        @test CloudStore.validate_container_name("ab-c1") == "ab-c1"
+        @test CloudStore.validate_container_name("a"^63) == "a"^63
+        @test CloudStore.validate_container_name("a"^3) == "a"^3
+    end
+
+    @testset "validate_key" begin
+        @test_throws ArgumentError CloudStore.validate_key("a"^1025)
+        @test_throws ArgumentError parse_s3_url(key="a"^1025)
+        @test_throws ArgumentError CloudStore.validate_key("a"^1026)
+        @test_throws ArgumentError parse_s3_url(key="a"^1026)
+
+        @test CloudStore.validate_key("a"^1024) == "a"^1024
+    end
+
+    @testset "validate_region" begin
+        @test_throws ArgumentError CloudStore.validate_region("no-region-1")
+        @test_throws ArgumentError parse_s3_url(region="no-region-1")
+
+        @test CloudStore.validate_key("us-east-1") == "us-east-1"
+    end
+
+    @testset "validate_blob" begin
+        @test_throws ArgumentError CloudStore.validate_blob("a"^1025)
+        @test_throws ArgumentError parse_azure_url(blob="a"^1025)
+        @test_throws ArgumentError CloudStore.validate_blob("a"^1026)
+        @test_throws ArgumentError parse_azure_url(blob="a"^1026)
+        @test_throws ArgumentError CloudStore.validate_blob(join(fill("a", 255), '/'))
+        @test_throws ArgumentError parse_azure_url(blob=join(fill("a", 255), '/'))
+        @test_throws ArgumentError CloudStore.validate_blob(join(fill("a", 256), '/'))
+        @test_throws ArgumentError parse_azure_url(blob=join(fill("a", 256), '/'))
+
+        @test CloudStore.validate_blob("a"^1024) == "a"^1024
+        @test CloudStore.validate_blob(join(fill("a", 254), '/')) == join(fill("a", 254), '/')
+    end
+
+    @testset "validate_account_name" begin
+        @test_throws ArgumentError CloudStore.validate_account_name("a")
+        @test_throws ArgumentError parse_azure_url(account="a")
+        @test_throws ArgumentError CloudStore.validate_account_name("aa")
+        @test_throws ArgumentError parse_azure_url(account="aa")
+        @test_throws ArgumentError CloudStore.validate_account_name("a"^25)
+        @test_throws ArgumentError parse_azure_url(account="a"^25)
+        @test_throws ArgumentError CloudStore.validate_account_name("a"^26)
+        @test_throws ArgumentError parse_azure_url(account="a"^26)
+        @test_throws ArgumentError CloudStore.validate_account_name("a.b")
+        @test !parse_azure_url(account="a.b")[1]
+        @test_throws ArgumentError CloudStore.validate_account_name("a-b")
+        @test_throws ArgumentError parse_azure_url(account="a-b")
+        @test_throws ArgumentError CloudStore.validate_account_name("a/b")
+        @test_throws ArgumentError parse_azure_url(account="a/b")
+
+        @test CloudStore.validate_account_name("abcd123456") == "abcd123456"
+        @test CloudStore.validate_account_name("1a1") == "1a1"
+    end
+end
```

Comment:
Similar comment: for all of the `validate*` functions, can you add a test with `""`? I assume it won't be valid, but maybe it would be actually? Anyway, it should have a test just to be sure. Thank you!